scripts/build_and_run_tests.bash

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+# Simple script to build swupd with multiple configurations and run all needed tests with different configurations
+
+set -e
+
+#configurable parameters
+JOB_COUNT=8
+CONFIGURE_ARGS="--with-fallback-capaths=$PWD/swupd_test_certificates --with-config-file-path=./testconfig --enable-third-party"
+
+export RUNNING_IN_CI=true
+run_build() {
+	make distclean || echo "No distclean - System already cleaned"
+	autoreconf -fi
+	./configure "$@"
+	make V=1 -j "${JOB_COUNT}"
+}
+
+run_checks() {
+	make -j "$JOB_COUNT" check
+	make -j "$JOB_COUNT" functional-check
+	make compliant
+	make shellcheck
+	make docs-coverage
+	make distcheck
+}
+
+# Run tests without any security sanitizer
+CFLAGS="" run_build --disable-optimizations "$CONFIGURE_ARGS"
+run_checks
+
+# Run build with basic configurations
+CFLAGS="$CFLAGS -fsanitize=address -fno-omit-frame-pointer -Werror" run_build "$CONFIGURE_ARGS"
+run_checks
```

Comment:
What is the purpose of running the tests twice once without security sanitizer and once with?

Reply:
This helps us to detect code problems that sanitizers and optimizations may hide. I remember we had a case in the past, but I don't remember what it is specifically.

Reply:
I would have thought that this combination would be enough:
- optimization **off**
- sanitizers **on**

But if you think there is value in running the tests twice changing those values I won't oppose. ☮️